### PR TITLE
tailcfg, control/controlclient, ipn/ipnlocal: add c2n (control-to-node) system

### PR DESCRIPTION
**Pull request by bradfitz**
This lets the control plane can make HTTP requests to nodes.

Then we can use this for future things rather than slapping more stuff
into MapResponse, etc.

### CODE REVIEW

**Comment by bradfitz** [comment]
(will add tests before marking it ready)

File: ipn/ipnlocal/c2n.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"io"
+	"net/http"
+)
+
+func (b *LocalBackend) handleC2N(w http.ResponseWriter, r *http.Request) {
+	switch r.URL.Path {
+	case "/hi":
+		io.WriteString(w, "hi\n")
```
Comment (maisem) [inline ipn/ipnlocal/c2n.go:15]: Let's do an echo instead, it helps test payloads too

Comment (bradfitz) [reply ipn/ipnlocal/c2n.go:15]: sure

File: ipn/ipnlocal/c2n.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	"io"
+	"net/http"
+)
+
+func (b *LocalBackend) handleC2N(w http.ResponseWriter, r *http.Request) {
+	switch r.URL.Path {
+	case "/echo":
+		// Test handler.
+		body, _ := io.ReadAll(r.Body)
+		w.Write(body)
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply ipn/ipnlocal/c2n.go:17]: thanks 🙄

**Comment by bradfitz** [comment]
Ready for review. cc @andrew-d

File: control/controlclient/direct.go [hunk 3]
```diff
@@ -1217,9 +1223,13 @@ func (c *Direct) answerPing(pr *tailcfg.PingRequest) {
 		c.logf("invalid PingRequest with no URL")
 		return
 	}
-	if pr.Types == "" {
+	switch pr.Types {
+	case "":
 		answerHeadPing(c.logf, httpc, pr)
 		return
+	case "c2n":
```
Comment (maisem) [inline control/controlclient/direct.go:1230]: return an error if `c2n` and `c.noiseConfigured == false`?

Comment (bradfitz) [reply control/controlclient/direct.go:1230]: I need this for tests. There's a TODO from you in our integration tests to make noise work with testcontrol

Comment (maisem) [reply control/controlclient/direct.go:1230]: I guess I should go fix that 😄

Comment (maisem) [reply control/controlclient/direct.go:1230]: can you add a env override for now?

**Review by maisem — APPROVED** [review]

**Pull request merged**